lib/block-supports/duotone.php

```diff
@@ -289,46 +272,21 @@ function gutenberg_render_duotone_support( $block_content, $block ) {
 		$duotone_values['b'][] = $color['b'] / 255;
 	}
 
-	$duotone_id = 'wp-duotone-filter-' . uniqid();
-
-	$selectors        = explode( ',', $duotone_support );
-	$selectors_scoped = array_map(
-		function ( $selector ) use ( $duotone_id ) {
-			return '.' . $duotone_id . ' ' . trim( $selector );
-		},
-		$selectors
-	);
-	$selectors_group  = implode( ', ', $selectors_scoped );
-
 	ob_start();
 
 	?>
 
-	<style>
-		<?php echo $selectors_group; ?> {
-			filter: url( <?php echo esc_url( '#' . $duotone_id ); ?> );
-		}
-	</style>
-
-	<svg
-		xmlns:xlink="http://www.w3.org/1999/xlink"
-		viewBox="0 0 0 0"
-		width="0"
-		height="0"
-		focusable="false"
-		role="none"
-		style="visibility: hidden; position: absolute; left: -9999px; overflow: hidden;"
-	>
+	<svg xmlns="http://www.w3.org/2000/svg">
```

Comment:
@jasmussen The nice thing about rendering the SVG in CSS is that we don't need all the tricks to hide it

Reply:
I'm catching up on this, sorry if I make you repeat things you already shared in other places!

So, one pro of embedding them in CSS is that we don't need to hide them. Another is that it can potentially be cached when/if we convert global styles to a stylesheet. One drawback is we can't target them via ID and use the `duotone: url(#wp-duotone-...)` in the theme.json. Can't think of anything else, though.

Reply:
One thing I've noticed is that the duotones set by the user output (repeated svgs):

```css
.wp-duotone-filter-613b790f42338 img {
	filter: url('data:image/svg+xml,<svg>...</svg>);
}
```

can we make it so it outputs (just points to the svg definition in global styles):

```css
.wp-duotone-filter-613b790f42338 img {
	filter: var(--wp-duotone-...);
}
```

Feel free to completely ignore this if it's too difficult, involved, or too long for this PR. We can address it in a follow-up if it's feasible. I presume it might require changes to how we store the block attributes for the duotone to tell apart custom from preset duotones, block migrations. etc.

Reply:
> So, one pro of embedding them in CSS is that we don't need to hide them. Another is that it can potentially be cached when/if we convert global styles to a stylesheet. One drawback is we can't target them via ID and use the `duotone: url(#wp-duotone-...)` in the theme.json. Can't think of anything else, though.

I guess they don't work in Safari, but the normal filters didn't work _correctly_ in Safari either. That seems like a big enough issue to potentially stop this method from proceeding. 

When initially building out the feature I gauged browser support off of [SVG filter](https://caniuse.com/svg-filters) support which said it was supported back to IE10. What I didn't see until @gwwar mentioned the Safari problem is [SVG effects in HTML](https://caniuse.com/svg-html) which is what we're trying to do and shows far less support.

> can we make it so it outputs (just points to the svg definition in global styles)
>
> Feel free to completely ignore this if it's too difficult, involved, or too long for this PR. We can address it in a follow-up if it's feasible.

Yeah, I can see that in a follow-up. I've started testing out generating `.has-<slug>-duotone-filter` classnames for the presets in https://github.com/WordPress/gutenberg/tree/add/duotone-has-filter-for-presets. Since the filter has to be applied to an inner element within a block, it wasn't going very well. I had a hack to set `--has-duotone-filter` and use `var(--has-duotone-filter)` somewhere within the block that's using duotone. Not exactly the best for developer experience considering it's currently "just add a line in your block.json" easy. Still generating the stylesheet and using the variable could be a slightly better alternative.